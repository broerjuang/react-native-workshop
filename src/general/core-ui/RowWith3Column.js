// @flow
import React from 'react';
import {View} from 'react-native';
type Props = {
  left?: ReactEl;
  content?: ReactEl;
  right?: ReactEl;
  style?: Object;
};

function RowWith3Column(props: Props) {
  let rightContent = React.isValidElement(props.right) ? props.right : null;

  let leftContent = React.isValidElement(props.left) ? props.left : null;

  let middleContent = React.isValidElement(props.content)
    ? props.content
    : null;

  let containerStyle = props.style ? props.style : [];
  return (
    <View>
      <View style={[styles.container, containerStyle]}>
        <View style={styles.edge}>{leftContent}</View>
        <View style={styles.middle}>{middleContent}</View>
        <View style={styles.edge}>{rightContent}</View>
      </View>
    </View>
  );
}

let widthConstantEdge = '15%';
let widthConstantMiddle = '70%';

let styles = {
  container: {
    flexDirection: 'row',
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: 'white',
  },
  middle: {
    padding: 10,
    width: widthConstantMiddle,
    justifyContent: 'center',
    alignItems: 'flex-start',
    flexWrap: 'wrap',
    // alignSelf: 'center',
  },
  edge: {
    width: widthConstantEdge,
    justifyContent: 'center',
    alignItems: 'center',
  },
};
export default RowWith3Column;
