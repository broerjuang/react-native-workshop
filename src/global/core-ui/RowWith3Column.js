// @flow
import React from 'react';
import {View, TouchableOpacity} from 'react-native';
type Props = {
  left?: ReactEl;
  content?: ReactEl;
  right?: ReactEl;
  style?: Object;
  isTouchable?: boolean;
  onPress?: Function;
};

function RowWith3Column(props: Props) {
  let {isTouchable, onPress} = props;
  isTouchable = isTouchable === undefined ? false : isTouchable;

  let rightContent = React.isValidElement(props.right) ? props.right : null;

  let leftContent = React.isValidElement(props.left) ? props.left : null;

  let middleContent = React.isValidElement(props.content)
    ? props.content
    : null;

  let containerStyle = props.style !== undefined ? props.style : {};
  return (
    <TouchableOpacity disabled={!isTouchable} onPress={onPress}>
      <View style={[styles.container, containerStyle]}>
        <View style={styles.edge}>{leftContent}</View>
        <View style={styles.middle}>{middleContent}</View>
        <View style={styles.edge}>{rightContent}</View>
      </View>
    </TouchableOpacity>
  );
}

let widthConstantEdge = '15%';
let widthConstantMiddle = '70%';

let styles = {
  container: {
    flexDirection: 'row',
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: 'white',
  },
  middle: {
    padding: 10,
    width: widthConstantMiddle,
    justifyContent: 'center',
    alignItems: 'flex-start',
    flexWrap: 'wrap',
    // alignSelf: 'center',
  },
  edge: {
    width: widthConstantEdge,
    justifyContent: 'center',
    alignItems: 'center',
  },
};
export default RowWith3Column;
