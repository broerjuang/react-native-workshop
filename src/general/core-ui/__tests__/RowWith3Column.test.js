import React from 'react';
import renderer from 'react-test-renderer';
import RowWith3Column from '../RowWith3Column';
import {View} from 'react-native';

describe('RowWith3Collumn', () => {
  let edge = <View style={{flex: 1, backgroundColor: 'blue'}} />;
  let middle = <View style={{flex: 1, backgroundColor: 'red'}} />;
  it('should render RowWith3Collumn corectly', () => {
    let component = renderer.create(
      <RowWith3Column left={edge} middle={middle} right={edge} />,
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
