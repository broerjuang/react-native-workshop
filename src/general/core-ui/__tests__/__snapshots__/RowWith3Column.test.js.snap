// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RowWith3Collumn should render RowWith3Collumn corectly 1`] = `
<View>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "white",
        "flexDirection": "row",
        "justifyContent": "center",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
          "width": "15%",
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "blue",
            "flex": 1,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "alignItems": "flex-start",
          "flexWrap": "wrap",
          "justifyContent": "center",
          "padding": 10,
          "width": "70%",
        }
      }
    />
    <View
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
          "width": "15%",
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "blue",
            "flex": 1,
          }
        }
      />
    </View>
  </View>
</View>
`;
